### PR TITLE
SW-3908 Changes to PR 1597 (warning dialog on update planting complete on map)

### PR DESCRIPTION
**Pull request by karthikbtf**
- refactored passing of id/val in a more constrained context
- moved reloadTracking into a useCallback so map doesn't go into an infinite rendering/reloading loop

### CODE REVIEW

**Comment by karthikbtf** [comment]
@constanzauanini this is on top of Charlie's PR.
Let's get charlie's PR in and he can review my changes and make a decision on top of what he has.

File: src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx [hunk 1]
```diff
@@ -63,10 +63,10 @@ export default function PlantingProgress({ reloadTracking }: PlantingProgressPro
     }
   }, [filters, activeLocale]);
 
-  const reloadTrackingAndObservations = () => {
+  const reloadTrackingAndObservations = useCallback(() => {
```
Comment (constanzauanini) [inline src/components/NurseryWithdrawals/PlantingProgressTabContent.tsx:66]: Cool!

**Comment by constanzauanini** [comment]
> @constanzauanini this is on top of Charlie's PR. Let's get charlie's PR in and he can review my changes and make a decision on top of what he has.

Ok, new changes looks good to me!

**Comment by karthikbtf** [comment]
> > @constanzauanini this is on top of Charlie's PR. Let's get charlie's PR in and he can review my changes and make a decision on top of what he has.
> 
> Ok, new changes looks good to me!

Thanks, I will merge Charlie's PR and leave this open for Charlie to review.
In the mean time, I will open another PR that makes reloadTracking a callback function -which we probably need for the list view as well.

**Pull request merged**